### ISSUE TITLE
Replace `Block->getSide()` with more versatile and less limited functions

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#6105]
## Description
`Block->getSide()` isn't really fit for purpose.
- A `Block` is required to use the method
- The method cannot be used without a `World` present, rendering it useless in `AsyncTask`s
- The `Block` must have been fetched from the world (requiring an unnecessary `Block` allocation and world read to access the API method)
- Alternatively, `getBlock(vector3->getSide(facing))` or e.g. `getBlock(vector3->up())` can be used, but that generates throwaway `Vector3` objects (or worse, `Position`s), and is also less readable and slower
- The function itself is slow due to needing to check `Position->isValid()` every time as well as repeated calls to `Position->getWorld()` which performs further validation checks
- It's a major obstacle to separating `Block`s and `Position`s, a change which would greatly improve system robustness, API design, and code readability. Current code involving block positions is awfully long after going from PM3 -> PM4.

TL;DR: This function is slow, inconvenient, and useless for any use-case that isn't within a `Block` descendent class itself and within a world.

While some (quite substantial) performance improvements have been made to the function in `minor-next`, there's still room for improvement.

A potentially better approach involves methods like the following:
- `ChunkManager->getBlockSide(Vector3 $pos, int $facing) : Block`
- `ChunkManager->getBlockSideAt(int $x, int $y, int $z, int $facing) : Block`

These functions would be able to avoid unnecessary extra allocations, `Position` objects, etc. while being convenient for world generators to use. Many existing usages would benefit from the performance advantage offered by such methods.

Perhaps `setBlockSide` equivalents could also be introduced, though I haven't seen a clear need for them so far. Getting blocks is done far more often than setting them.

## Justification
Mainly improving performance. These new functions don't allow anything that wasn't already possible. They just allow doing some existing tasks with less waste.

**Comment by Bqleine** [comment]
I think `setBlockSide` equivalents should also be made simply to maintain consistency and not have a different syntax to set and get the same thing.
